src/main/java/org/manifold/compiler/front/NodeValueVertex.java

```diff
@@ -0,0 +1,226 @@
+package org.manifold.compiler.front;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.manifold.compiler.ConnectionValue;
+import org.manifold.compiler.InvalidAttributeException;
+import org.manifold.compiler.NodeTypeValue;
+import org.manifold.compiler.NodeValue;
+import org.manifold.compiler.PortTypeValue;
+import org.manifold.compiler.PortValue;
+import org.manifold.compiler.TypeMismatchException;
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.UndeclaredAttributeException;
+import org.manifold.compiler.UndeclaredIdentifierException;
+import org.manifold.compiler.UndefinedBehaviourError;
+import org.manifold.compiler.Value;
+
+public class NodeValueVertex extends ExpressionVertex {
```

Comment:
All that being said, this battle probably isn't worth fighting because we just want to have a demo for the end of the semester. :person_frowning: I'm frustrated that it has been hard to refactor our high level architecture when necessary within these 1000+ line PRs.